micromagneticdata/drive.py

```diff
@@ -415,3 +416,29 @@ def slider(self, description="step", **kwargs):
         return ipywidgets.IntSlider(
             value=0, min=0, max=self.n - 1, step=1, description=description, **kwargs
         )
+
+    def to_xarray(self):
+        """Export ``micromagneticdata.Drive`` as ``xarray.Dataset``"""
+        if self.info["driver"] == "MinDriver":
+            darray = self[0].to_xarray()
+        else:
+            field_darrays = (field.to_xarray() for field in self)
+            if self.info["driver"] == "TimeDriver":
+                darray = xr.concat(field_darrays, dim=self.table.data["t"])
+                darray.t.attrs["units"] = self.table.units["t"]
+            else:
+                darray = xr.concat(field_darrays, dim="B_hysteresis").assign_coords(
+                    {
+                        f"B{i}_hysteresis": (
+                            "B_hysteresis",
+                            self.table.data[f"B{i}_hysteresis"],
+                        )
+                        for i in "xyz"
+                    }
+                )
+                for i in "xyz":
+                    darray[f"B{i}_hysteresis"].attrs["units"] = self.table.units[
+                        f"B{i}_hysteresis"
+                    ]
```

Comment:
Maybe we could combine the two for loops and assign the coordinates in the second loop?

Reply:
Yes, I am not happy with two for loops. Ideally I would like to add the coordinates and attributes to those coordinates in the same line; hence I am trying a few things. If nothing works, I am thinking of one for loop where we add a coordinate in the first line and then its `units` attribute in the second line, for all x, y, and z components.